### ISSUE TITLE
Scoring

### ISSUE THREAD

**Issue by Limpan** [Limpan/parsons-puzzle#13]
It would be possible to score the users performance.

**Comment by Greenheart** [comment]
## Decreasing reward
This builds on the idea that puzzles have a difficulty stored as a number between 1 and 100
* When a user starts a challenge, a variable with the reward is set to the puzzle difficulty's value (Ex. 73).
* For each failed try, the reward is decrease by some amount.
  * Maybe decrease the reward by larger amounts when the reward is high
* The reward can't get smaller than let's say 1 or 5 out of 100.
* When the user submits a valid solution, they get the reward.
  * But only if they haven't already completed the challenge
    * Maybe store an user's completed challenges as an dictionary with the `challengeId`s as keys and `received_reward` as values
      * `User.completed_challenges = {"challenge_id": received_reward}`
      * Ex: {"5": 18} - this user has completed challenge 5 and got 18 points.
      * Also keep a `User.total_score` to not have to compute this over and over if users at some point get hundreds of completed challenges.

## Why this approach?
* Gamification
* Users get to gather points - could be showed in `a profile` and/or `a leaderboard` of some kind?
* By decreasing the reward on each submitted try, users learn that they need to be thoughtful about their guesses. This makes them think deeper about each attempt - which hopefully helps them learn more from each puzzle. Extreme example: Users don't learn much by mindlessly bruteforcing combinations and still getting points.

**Comment by Limpan** [comment]
I agree on the gamification of things. It seems to be a rather effective approach. It is a future feature but it should of course be somewhat considered for the database design.
